plugins/Dashboard/Dashboard.php

```diff
@@ -32,10 +32,21 @@ public function registerEvents()
             'Translate.getClientSideTranslationKeys' => 'getClientSideTranslationKeys',
             'Widget.addWidgetConfigs'                => 'addWidgetConfigs',
             'Category.addSubcategories'              => 'addSubcategories',
-            'Widgetize.shouldEmbedIframeEmpty'       => 'shouldEmbedIframeEmpty'
+            'Widgetize.shouldEmbedIframeEmpty'       => 'shouldEmbedIframeEmpty',
+            'Db.getTablesInstalled'                  => 'getTablesInstalled'
         );
     }
 
+    /**
+     * Register the new tables, so Matomo knows about them.
+     *
+     * @param array $allTablesInstalled
+     */
+    public function getTablesInstalled(&$allTablesInstalled)
+    {
+        $allTablesInstalled[] = Common::prefixTable('user_dashboard');
```

Comment:
@sgiehl is it possible these tables are installed even when the plugin is deactivated? It wouldn't find this table then? Not sure we can do much about this? guess for core plugins we could at least have them all in corehome for example but would still be an issue for other plugins. Not sure where we call the getDbInstalled event and whether we can in these cases temporarily load all plugins or not. (might cause other issues). Not sure there's a solution to it?

Reply:
Yes it can happen, that those tables are present, but the plugin is not enabled. On the other hand it can happen that the tables are not yet created, so moving them to corehome might cause other issues. But that also applies for all additional plugins one has installed.
I've adjusted the method so it loads all plugins before posting the event.

Reply:
Went through all usages of the method and might be fine to load all plugins. However, `ArchiveTablesCreator::createArchiveTablesIfAbsent()` might be a problem? The `getNumericTable` and `getBlobTable` seem to cause getting a list of all tables causing maybe a performance regressions. Does methods are used very often in many places and there could be performance regressions. Eg every time we load an archive we might fetch all tables etc. Could you double check this and potentially change the `ArchiveTableCreator` class to not use regular `getTablesInstalled` method or so? Maybe it just needs to check manually for a list of all `Common::prefixTable('archive_')` tables and use this list? To avoid loading and unloading plugins and also to avoid triggering event etc.

Reply:
implemented a simple new method in ArchiveTablesCreator to only fetch the archive tables.